tesseract/tesseract_environment/test/tesseract_environment_unit.cpp

```diff
@@ -65,8 +86,28 @@ void runContactManagerCloneTest(const tesseract_environment::Environment::Ptr& e
   EXPECT_TRUE(std::equal(e_active_list.begin(), e_active_list.end(), c_active_list.begin()));
 }
 
-void runAddandRemoveLinkTest(const tesseract_environment::Environment::Ptr& env)
+template <typename S>
+void runAddandRemoveLinkTest()
 {
+  SceneGraph::Ptr scene_graph = getSceneGraph();
+  EXPECT_TRUE(scene_graph != nullptr);
+
+  auto env = std::make_shared<Environment>();
+  EXPECT_TRUE(env != nullptr);
+
+  bool success = env->init<S>(*scene_graph);
+  EXPECT_TRUE(success);
+
+  // Register contact manager
+  EXPECT_TRUE(env->registerDiscreteContactManager(tesseract_collision_bullet::BulletDiscreteBVHManager::name(),
+                                                  &tesseract_collision_bullet::BulletDiscreteBVHManager::create));
+  EXPECT_TRUE(env->registerContinuousContactManager(tesseract_collision_bullet::BulletCastBVHManager::name(),
+                                                    &tesseract_collision_bullet::BulletCastBVHManager::create));
+
+  // Set Active contact manager
+  EXPECT_TRUE(env->setActiveDiscreteContactManager(tesseract_collision_bullet::BulletDiscreteBVHManager::name()));
+  EXPECT_TRUE(env->setActiveContinuousContactManager(tesseract_collision_bullet::BulletCastBVHManager::name()));
```

Comment:
This code is repeated a few times. Consider putting in a utility

Reply:
Done.